system-tests/lib/cre/contracts/workflowregistry.go

```diff
@@ -0,0 +1,263 @@
+package contracts
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/jmoiron/sqlx"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/scylladb/go-reflectx"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	cldf_chain "github.com/smartcontractkit/chainlink-deployments-framework/chain"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	"github.com/smartcontractkit/chainlink-evm/pkg/logpoller"
+	"github.com/smartcontractkit/chainlink-testing-framework/framework/components/postgres"
+
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	ks_contracts_op "github.com/smartcontractkit/chainlink/deployment/keystone/changeset/operations/contracts"
+
+	libc "github.com/smartcontractkit/chainlink/system-tests/lib/conversions"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/don/node"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/flags"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/infra"
+)
+
+// must match nubmer of events we track in core/services/workflows/syncer/handler.go
+const NumberOfTrackedWorkflowRegistryEvents = 6
+
+func WaitForWorkflowRegistryFiltersRegistration(
+	testLogger zerolog.Logger,
+	singleFileLogger logger.Logger,
+	infraType infra.Type,
+	registryChainID uint64,
+	topology *cre.DonTopology,
+	nodeSetInput []*cre.CapabilitiesAwareNodeSet,
+) error {
+	// we currently have no way of checking if filters were registered, when code runs in CRIB
+	// as we don't have a way to get its database connection string
+	if infraType == infra.CRIB {
+		return nil
+	}
+
+	hasGateway := false
```

Comment:
💥 Now this is some code I did not expect

Why is this relevant? What breaks if we remove it?